indigo/indigo-extras/src/main/scala/indigoextras/waypoints/WaypointPath.scala

```diff
@@ -0,0 +1,113 @@
+package indigoextras.waypoints
+
+import indigo.Radians
+import indigo.Vector2
+
+import scala.annotation.tailrec
+
+trait WaypointPath:
+  def waypoints: List[Vector2]
+
+  def effectiveWaypoints: List[Vector2]
+
+  def loop: Boolean
+
+  def calculatePosition(
+      at: Double
+  ): (Vector2, Radians)
```

Comment:
Replace the Tuple return type with a nice little case class with named arguments.